go/vt/vttablet/tabletserver/planbuilder/dml.go

```diff
@@ -323,6 +323,12 @@ func analyzeInsert(ins *sqlparser.Insert, tables map[string]*schema.Table) (plan
 	}
 	table, tableErr := plan.setTable(tableName, tables)
 
+	// message inserts need to continue being strict, even in passthrough dml mode,
+	// because field defaults are set here
+	if tableErr == nil && table.Type == schema.Message {
```

Comment:
I think we should never return a default plan if table type is Message. So, we require the following changes:

If tableErr is not nil, it will end up returning the pass-through plan. This check should instead be:
```
if table.Type == schema.Message {
  if tableErr != nil {
    return nil, tableErr
  }
  return analizeInsertMessage(...)
}
```

The change also skips the `HasPrimary` check. There is a re-check in the `analyzeInsertMessage` for it (look for Dead code comment).

That should be changed to return an error instead.

We also need tests that demonstrate the new behavior. I can think of three cases:
1. Show that we error out if tableErr != nil even if PassthroughDMLs is true.
2. Show that plan is being built even if PassthroughDMLs is true.
3. Show that we generate an error if table has no PK.

Reply:
Actually, if tableErr != nil, then table.Type will never be set.
So, ignore my first suggestion.

That should also reduce the number of tests to two.

Reply:
I just refactored the logic so that #2 is the only new behavior. It's a much more readable change with fewer branching paths. Tests now included.